src/registerServiceWorker.js

```diff
@@ -0,0 +1,117 @@
+// In production, we register a service worker to serve assets from local cache.
```

Comment:
This file is not the latest and greatest from create-react-app.

Reply:
Hm. I started with CRA 1.x and followed the instructions for migrating to 2.0. I'll see if I can do better.

Reply:
Hm. I started with CRA 1.x and followed the instructions for migrating to 2.0. I'll see if I can do better.

Reply:
I don't think the serviceworker gets upgrades when you upgrade because the file is part of the `src` directory and dumped from the template into the new directory. 

The new hotness (and it's hot!) is called `serviceWorker.js` and the only way to get it, as far as I know, is to create a throw-away cra app in `/tmp` and copy the `src/serviceWorker.js` file from there. And then to also add...:

```javascript
import * as serviceWorker from './serviceWorker';
```
and at the bottom:
```javascript
serviceWorker.register({
  onUpdate: () => {
    window.location.reload()
  }
});
```